### ISSUE TITLE
commandLineArgs not working when adding a project with a launchProfileName

### ISSUE THREAD

**Issue by distantcam** [dotnet/aspire#6892]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

When adding a project, and specifying a `launchProfileName` the correct launch profile is used. However the `commandLineArgs` of the launch profile are not being passed on during execution.

### Expected Behavior

The `commandLineArgs` are passed through as the arguments when the project is run.

### Steps To Reproduce

https://github.com/distantcam/AspireLaunchArgsIssue

Run the apphost, notice the prestart resource doesn't finish, blocking the start of the main resource.

### Anything else?

This was working in the previous version of Aspire (8.0).

**Comment by mitchdenny** [comment]
Actually may be on the app model side should check that first.

**Comment by davidfowl** [comment]
I ran this sample using Aspire 9 with vs code and dotnet run and it works flawlessly 😄 . Will try vs next

**Comment by davidfowl** [comment]
Reproduces in visual studio. @bradygaster @JakeRadMSFT can you take a look at this one. That inconsistent behavior is a problem.

**Comment by davidfowl** [comment]
cc @karolz-ms

**Comment by LittleLittleCloud** [comment]
After debugging into Aspire service from VS side, it seems that the `prestart` profile is propagating between DCP and VS correctly. 
Moreover, If I change `AspireLaunchArgIssue` project by sleeping 10 seconds in `prestart` profile, this project would work in VS. 
![Image](https://github.com/user-attachments/assets/f71bd662-62c1-4194-823d-4d6e8f21f6c7)

Therefore I don't think this issue is related to how `commandLineArgs` or `launchProfileName` being processed when launching target in IDE. My guess for the cause of this issue is the `AspireLaunchArgIssue` ends too quickly after its start, which cause either DCP or VS fails to get notified when the process is completed.

Will keep investigating with @karolz-ms for a solution.

**Comment by davidfowl** [comment]
@LittleLittleCloud Are you using the latest internal build of VS? I should add that I reproduced on public preview 😄

**Comment by distantcam** [comment]
I updated the example to include the delay, as well as outputting the arguments to the console. I'm currently using VS 17.12.3.

```csharp
Console.WriteLine($"Args: '{string.Join(", ", args)}'");

if (args.Length > 0 && args[0] == "prestart")
{
    // Do prestart logic
    await Task.Delay(10_000);
    return;
}

var builder = WebApplication.CreateBuilder(args);

builder.AddServiceDefaults();
var app = builder.Build();

app.MapDefaultEndpoints();

app.MapGet("/", () => "Hello World!");

app.Run();
```

![Image](https://github.com/user-attachments/assets/f35d853a-25ea-4c1e-9309-907c52fe40fd)

**Comment by davidfowl** [comment]
Can you dump all of the arguments?

**Comment by distantcam** [comment]
I'm doing `Console.WriteLine($"Args: '{string.Join(", ", args)}'");` inside a top level statement. I'm not sure what other args you're refering to.

**Comment by distantcam** [comment]
When run through `dotnet run`

![Image](https://github.com/user-attachments/assets/e0c0d058-415d-4f0b-b108-e72261a1dbb1)

**Comment by davidfowl** [comment]
Sorry I meant @LittleLittleCloud 😄

**Comment by LittleLittleCloud** [comment]
@davidfowl Yes I was verifying on the latest internal build of VS,, to be specific, the below version.
```
Version 17.14.0 Preview 1.0 [35726.43.main]
```

After delaying 10 seconds, the prestart resource no longer block, and it appears to behave normal. The console output of prestart resource is below on my end

![Image](https://github.com/user-attachments/assets/3abfff15-b312-4795-b109-f9291b1ec48c)

 Let me verify it on latest released VS (17.12.4) && VS preview and see if this issue is reproducable there

**Comment by karolz-ms** [comment]
Thanks @LittleLittleCloud 

Regardless what you find, there should be no case where a project quitting "very fast" is causing issues. If there is a timing issue in the current implementation, we have to find and fix it. E.g. in the past I have been working with @tlmii to properly handle the case where project termination notification arrives before the request to start THAT VERY PROJECT completes.

**Comment by joperezr** [comment]
[From Work item meeting] - It'll be interesting to check if this is fixed in the latest version of Visual Studio as well as VSCode. @bradygaster will take a look.

**Comment by pflannery** [comment]
I have the same problem. 

I found this [comment line](https://github.com/dotnet/aspire/blob/3eebc370ea9503aeed1430c9aa0a46b4187a01d7/src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs#L478) states that the command line args will be processed by `ApplicationExecutor.PrepareProjectExecutables` but in that function these [if statements](https://github.com/dotnet/aspire/blob/4bf0c657521d5434e92a38c1ba43df19f729baea/src/Aspire.Hosting/Dcp/DcpExecutor.cs#L816) never actually add any command arguments when I debug through. 

This [if branch](https://github.com/dotnet/aspire/blob/4bf0c657521d5434e92a38c1ba43df19f729baea/src/Aspire.Hosting/Dcp/DcpExecutor.cs#L863) has logic to add the launch profile command line arguments but it also inserts a "--" argument before the launchProfile commandLineArgs which wasn't what I was expecting.

I thought maybe it would be simpler to just add an additional [ResourceBuilderExtensions](https://github.com/dotnet/aspire/blob/3eebc370ea9503aeed1430c9aa0a46b4187a01d7/src/Aspire.Hosting/ResourceBuilderExtensions.cs#L18) function like this instead:

```cs
builder.AddProject<SomeApp>("some-app", launchProfileName: "some-profile")
    .WithLaunchProfileArgs();
```

Working source code:

```cs
    public static IResourceBuilder<T> WithLaunchProfileArgs<T>(this IResourceBuilder<T> builder) where T : IResourceWithArgs
    {
        ArgumentNullException.ThrowIfNull(builder);

        var project = builder.Resource as ProjectResource;
        ArgumentNullException.ThrowIfNull(project);

        var launchProfile = project.GetEffectiveLaunchProfile()?.LaunchProfile;
        if (launchProfile is null)
        {
            ArgumentNullException.ThrowIfNull(launchProfile);
        }

        if (string.IsNullOrWhiteSpace(launchProfile.CommandLineArgs))
        {
            ArgumentNullException.ThrowIfNull(launchProfile.CommandLineArgs);
        }

        var cmdArgs = CommandLineArgsParser.Parse(launchProfile.CommandLineArgs);

        return builder.WithArgs(cmdArgs.ToArray());
    }
```

**Comment by danmoseley** [comment]
@LittleLittleCloud if this is fixed, lets close it.
